### PR TITLE
PowerDNS: Implemented AUTODNSSEC

### PR DESCRIPTION
**Pull request by jpbede**
This PR implements AUTODNSSEC for PowerDNS

### CODE REVIEW

File: providers/powerdns/dnssec.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// getDNSSECCorrections returns corrections that update a domain's DNSSEC state.
+func (api *PowerDNS) getDNSSECCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	cryptokeys, getErr := api.client.Cryptokeys().ListCryptokeys(context.Background(), api.ServerName, dc.Name)
+	if getErr != nil {
+		return nil, getErr
+	}
+
+	// check if any of the avail. key is active and published
+	hasEnabledKey := false
+	keyID := 0
```
Comment (tlimoncelli) [inline providers/powerdns/dnssec.go:18]: Does the zero have any significance?  If not, I suggest: `var keyID int`

Comment (jpbede) [reply providers/powerdns/dnssec.go]: No, fixed that. Thanks for the hint.

**Comment by tlimoncelli** [comment]
Looks good so far.  Is this ready for ready for merge? (and final review)

**Comment by jpbede** [comment]
Yep. This works for me, it disables/enables DNSSEC for my domains.

**Comment by patschi** [comment]
Just to be sure: What's happening with existing PowerDNS configurations where `AUTODNSSEC` is not set? Wouldn't this cause DNSSEC to be disabled?

**Comment by jpbede** [comment]
Yes, that is right.

**Comment by patschi** [comment]
I'd see this as a as it could break or cause major issues with existing deployments. It might be a good idea delaying this change until AUTODNSSEC has been rewritten (see #847), wouldn't it @tlimoncelli? Just my two cents.

**Comment by tlimoncelli** [comment]
I was assuming nobody would be using AUTODNSSEC thus #847 wasn't that urgent.  I now realize was a bad assumption.

None of the providers I have access to support AUTODNSSEC, which makes it difficult for me to write that code.  Can someone else take a stab at it?

**Comment by patschi** [comment]
> Can someone else take a stab at it?

If my suggested way-to-go in #847 is fine, I **might** invest some time next week or so. However: no promises, as it depends a lot how motivated for doing Go-stuff :D

> The workaround is [...] you'll need to be explicit about whether AUTODNSSEC is on or off when you upgrade...

To be honest this change should not be included in a release with that "workaround" without reworking AUTODNSSEC previously. I can imagine this might cause quite some troubles for who's using PowerDNS...

**Comment by tlimoncelli** [comment]
I have some time today to make a draft PR.  I'll need others to test it.

Luckily since Go is statically typed, I can just change this one field and the compiler will tell me what to change.

**Comment by jpbede** [comment]
I can have a look and test the PR at the mid./end of the next week.

**Comment by tlimoncelli** [comment]
I made this: https://github.com/StackExchange/dnscontrol/pull/863

I changed the name of AUTODNSSEC_ON and AUTODNSSEC_OFF to be explicit.  Including neither makes no changes.  (Anyone using the old "AUTODNSSEC" keyword will fail. I consider that a good thing since we've changed behavior).

Dumb question but.... Does anyone ever turn AUTODNSSEC off?  Once you turn it on, wouldn't you leave it on forever?  In that case, instead of that PR, we could just remove the code that disables it.

**Pull request merged**